src/backend/gl/gl_common.c

```diff
@@ -35,11 +35,18 @@ struct gl_blur_context {
 	/// Temporary textures used for blurring. They are always the same size as the
 	/// target, so they are always big enough without resizing.
 	/// Turns out calling glTexImage to resize is expensive, so we avoid that.
-	GLuint blur_texture[2];
-	/// Temporary fbo used for blurring
-	GLuint blur_fbo;
-
-	int texture_width, texture_height;
+	GLuint *blur_textures;
+	int blur_texture_count;
+	/// Temporary fbos used for blurring
+	GLuint *blur_fbos;
+	int blur_fbo_count;
+	/// Cached size of each blur_texture
+	struct texture_size {
+		int width;
+		int height;
+	} * texture_sizes;
+
+	int fb_width, fb_height;
```

Comment:
Can you add a comment about what these variables are for? I couldn't figure it out just by the name.

Reply:
Will do. I'll also have to take a look again at those new `clang-tidy` warnings.

Reply:
Note with the new `.clang-tidy`, it doesn't run clang-analyzer checks anymore. But meson lets you run them easily with `ninja scan-build`

Reply:
Please take a look at the updated comments. We could probably get away with leaving `fb_*` out but this would require some assumptions about the `texture_sizes` array (at least one entry, first one should be equal to the current `fb_*` value) that aren't obvious and not given at the moment.

`texture_sizes` is only really needed by the dual-filter kawase algorithm, as for the kernel blur these are equal to `fb_width`/`fb_height`. These can probably also be calculated on the fly, but I think it's cleaner to explicitly cache them than to calculate them at multiple places.